Add is_leaf argument to pytrees\pytrees.py with test at tests\pytrees\test_pytrees.py, fixes #6083 (#6107)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  relaxation to occur during longer measurements. The trit flip error represents misclassification during readout.
  [(#5842)](https://github.com/PennyLaneAI/pennylane/pull/5842)

> * Added `is_leaf` parameter to function `flatten` in `pytrees.pytrees.py`. This is to allow node flattening to be stopped for any node where the `is_leaf` Optional argument evaluates to being true.

Reviewer: ```suggestion
* Added `is_leaf` parameter to function `flatten` in the `qml.pytrees` module. This is to allow node flattening to be stopped for any node where the `is_leaf` optional argument evaluates to being `True`.
```

>   [(#6083)](https://github.com/PennyLaneAI/pennylane/issues/6083)

Reviewer: The link should be to the PR, not to the issue 😄 

```suggestion
  [(#6107)](https://github.com/PennyLaneAI/pennylane/issues/6107)
```

Guillermo Alonso,
Utkarsh Azad,

> Gabriel Bottrill,
> Tonmoy T. Bhattacharya,

Reviewer: ```suggestion
Tonmoy T. Bhattacharya,
Gabriel Bottrill,
```
